src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp

```diff
@@ -36,12 +36,10 @@ TEST_F(TestEnlsvgPathPlanner, test_enlsvg_path_planner_empty_grid)
 
     EXPECT_TRUE(path != std::nullopt);
```

Comment:
unrelated nit: this should be ASSERT_TRUE so it stops before trying to access a nullopt

Reply:
I've made the change, but I'm not sure I entirely understand it. What's the difference between the EXPECT_TRUE and ASSERT_TRUE in this scenario?

Reply:
They are essentially testing the same thing (that two values are equal to each other). If EXPECT fails, execution continues to the next line but with ASSERT, execution stops at that point. We should be doing ASSERT here (which I overlooked when I wrote these tests) because we don't want to access a std::nullopt because that's essentially a null pointer access